src/plugins/presentation_panel/public/panel_component/panel_header/presentation_panel_hover_actions.tsx

```diff
@@ -0,0 +1,563 @@
+      canceled = true;
+      subscriptions.unsubscribe();
+    };
+  }, [api, quickActionIds]);
+
+  useEffect(() => {
+    if (!api) return;
+
+    let canceled = false;
+    const apiContext = { embeddable: api };
+
+    (async () => {
+      let compatibleActions = (await (async () => {
+        if (getActions) return await getActions(CONTEXT_MENU_TRIGGER, apiContext);
+        return (
+          (await uiActions.getTriggerCompatibleActions(CONTEXT_MENU_TRIGGER, {
+            embeddable: api,
+          })) ?? []
+        );
+      })()) as AnyApiAction[];
+      if (canceled) return;
+
+      const disabledActions = api.disabledActionIds?.value;
+      if (disabledActions) {
+        compatibleActions = compatibleActions.filter(
+          (action) => disabledActions.indexOf(action.id) === -1
+        );
+      }
+
+      if (actionPredicate) {
+        compatibleActions = compatibleActions.filter(({ id }) => actionPredicate(id));
+      }
+
+      compatibleActions.sort(
+        ({ order: orderA }, { order: orderB }) => (orderB || 0) - (orderA || 0)
+      );
+
+      const contextMenuActions = compatibleActions.filter(
+        ({ id }) => !(quickActionIds as readonly string[]).includes(id)
+      );
+
+      const menuPanels = await buildContextMenuForActions({
+        actions: contextMenuActions.map((action) => ({
+          action,
+          context: apiContext,
+          trigger: contextMenuTrigger,
+        })) as ActionWithContext[],
+        closeMenu: onClose,
+      });
+      setContextMenuPanels(menuPanels);
+      setShowNotification(contextMenuActions.some((action) => action.showNotification));
+      setQuickActions(
+        compatibleActions.filter(({ id }) => (quickActionIds as readonly string[]).includes(id))
+      );
+    })();
+
+    return () => {
+      canceled = true;
+    };
+  }, [
+    actionPredicate,
+    api,
+    getActions,
+    isContextMenuOpen,
+    onClose,
+    parentViewMode,
+    quickActionIds,
+  ]);
+
+  const quickActionElements = useMemo(() => {
+    if (!api || quickActions.length < 1) return [];
+
+    const apiContext = { embeddable: api, trigger: contextMenuTrigger };
+
+    return quickActions
+      .sort(({ order: orderA }, { order: orderB }) => {
+        const orderComparison = (orderB || 0) - (orderA || 0);
+        return orderComparison;
+      })
+      .map((action) => {
+        const name = action.getDisplayName(apiContext);
+        const iconType = action.getIconType(apiContext) as IconType;
+        const id = action.id;
+
+        return {
+          iconType,
+          'data-test-subj': `embeddablePanelAction-${action.id}`,
+          onClick: createClickHandler(action, apiContext),
+          name,
+          id,
+        };
+      });
+  }, [api, quickActions]);
+
+  const notificationElements = useMemo(() => {
+    if (!showNotifications || !api) return [];
+    return notifications?.map((notification) => {
+      console.log('notificationElements', { api });
```

Comment:
@cqliu1 I think this console.log statement is clobbering the journeys test. Removing it on my machine fixed the test. 🤞